core/src/stored/stored_conf.cc

```diff
@@ -736,6 +736,11 @@ static bool SaveResource(int type, ResourceItem* items, int pass)
   int i;
   int error = 0;
 
+  BareosResource* allocated_resource = *resources[rindex].allocated_resource_;
+  if (pass == 2 && !allocated_resource->Validate()) {
+    return false;
```

Comment:
Does this return value mean that the storage daemon does not start?

Reply:
Yes. If Validate() returns false, the the SD will not start. This is analogous to the behaviour of the Director.
The idea is that we can run in-depth checks of the configuration that way (which was why that function added to the director in the first place).